tests/jax/core_system_components/system_component_test.py

```diff
@@ -282,6 +283,26 @@ def test_system_launch_with_build(
     assert system_with_two_components._builder.store.int_plus_str == 3
     assert system_with_two_components._builder.store.float_plus_bool == 1.2
     system_with_two_components.launch()
+    assert system_with_two_components._builder.store.global_config == SimpleNamespace(
+        multi_process=True,
```

Comment:
Good catch. Sorted. It now tests for the defaults from the the defaults dataclasses unless the parameters are overwritten. Just one thing to note, I have to instantiate the `DistributorDefaultConfig` in order to get access to the `nodes_on_gpu` attribute. I think it is because its default is given using `field`. Happy to change it to just check for an empty list instead.